lib/create-contentful-api.ts

```diff
@@ -482,12 +483,22 @@ export default function createContentfulApi<OptionType>(
     id: string,
     query: EntryQueries = {}
   ): Promise<EntryWithLinkResolution<Fields>> {
+    if (query.locale === '*') {
+      console.warn(
+        'If you want to fetch entries in all existing locales, we recommend you to use client.withAllLocales'
```

Comment:
[sorry for the long text, it's just there to explain my suggested change ✏️ ]

Just thought about this warning and the error below. I think at the moment we risk the flow being:

1/ user uses `client.getEntries({locale: '*'})`
2/ warning: we recommend you to use `client.withAllLocales`
3/ user changes to `client.withAllLocales.getEntries({locale: '*'})`
4/ error: locale not allowed
5/ user is confused about mixed messages

To mitigate I think we can make the warning more specific so when people see it in the console they can go back to the code and know what to change:

```suggestion
        'If you want to fetch an entry in all existing locales, we recommend you to use client.withAllLocales instead of the locale='*' parameter'
```

Reply:
I agree, great example to show me the confusion :)